### PR TITLE
[ADD][14.0] new addon contract_sequence

### PR DESCRIPTION
**Pull request by scigghia**
This addon adds sequence to contract and contract template
it sets also a default sequence for contract in company settings

### CODE REVIEW

**Comment by angelmoya** [comment]
Tested on runboat, I can not create a customer contract.

**Comment by scigghia** [comment]
thank you for your feedback @angelmoya i fixed some default values and now should be fine
but i think runboat for this pr is broken

**Comment by scigghia** [comment]
@sbidoul could you take a look to the runboat please?

**Comment by sbidoul** [comment]
likely due to this: https://github.com/OCA/oca-ci/issues/29

**Comment by scigghia** [comment]
@angelmoya did you have the opportunty to test? now the runboat is running :)

**Comment by angelmoya** [comment]
Nice, is working right... but codecov is to low.. if you can add tests would be great

**Comment by scigghia** [comment]
why travis and runboat are crashing on account_banking_mandate ?
@sbidoul

**Comment by sbidoul** [comment]
That was an expired certificate on odoo-community.org again. It's resolved now please retry.

**Comment by scigghia** [comment]
> That was an expired certificate on odoo-community.org again. It's resolved now please retry.

thank you @sbidoul

**Comment by scigghia** [comment]
no one interested in this module?

File: contract_sequence/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 Andrea Cometa - Apulia Software (www.apuliasoftware.it)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Contract Sequence",
+    "version": "14.0.0.0.1",
```
Comment (rousseldenis) [inline contract_sequence/__manifest__.py]:
```suggestion
    "version": "14.0.1.0.0",
```

File: contract_sequence/models/contract.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "to be proposed.",
+        copy=False,
+    )
+
+    def get_default_sequence(self):
+        result = self.env.company.contract_default_sequence or self.env.ref(
+            "contract_sequence.seq_contract_auto"
+        )
+        if not result:
+            raise UserWarning(_("No default sequence found!"))
```
Comment (rousseldenis) [inline contract_sequence/models/contract.py:24]: Please raise Odoo exception (UserError)

File: contract_sequence/models/contract.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        return result
+
+    sequence_id = fields.Many2one(
```
Comment (rousseldenis) [inline contract_sequence/models/contract.py:27]: Please put fields definition on top

File: contract_sequence/models/contract.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        comodel_name="ir.sequence",
+        string="Contract Sequence",
+        help="This field contains the information related to the numbering "
+        "of the journal entries of this journal.",
+        copy=False,
+        default=get_default_sequence,
```
Comment (rousseldenis) [inline contract_sequence/models/contract.py:33]: Please use lambda

File: contract_sequence/models/contract_template.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Andrea Cometa - Apulia Software (www.apuliasoftware.it)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class ContractTemplate(models.Model):
+    _inherit = "contract.template"
```
Comment (rousseldenis) [inline contract_sequence/models/contract_template.py:8]: Is the sequencing wanted on templates ?

File: contract_sequence/tests/test_contract_sequence.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
+# Copyright 2022 Andrea Cometa - Apulia Software (www.apuliasoftware.it)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase, tagged
+
+from ..hooks import pre_init_hook
+
+
+@tagged("post_install", "-at_install")
+class TestContractSequence(TransactionCase):
```
Comment (rousseldenis) [inline contract_sequence/tests/test_contract_sequence.py:10]: Please use SavepointCase

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Pull request closed without merging**